tests/unit/ipCheck.js

```diff
@@ -0,0 +1,76 @@
+'use strict'; // eslint-disable-line strict
+
+const assert = require('assert');
+const ipCheck = require('../../lib/ipCheck');
+const ipaddr = require('ipaddr.js');
+
+function parseValidIpCheck(ip, sup) {
+    const actualRes = ipCheck.parseIp(ip);
+    assert(actualRes instanceof sup);
+}
+
+function parseInvalidIpCheck(ip) {
+    const actualRes = ipCheck.parseIp(ip);
+    assert.deepStrictEqual(actualRes, {});
+}
+
+function cidrMatchCheck(cidr, ip, expectedRes) {
+    const actualRes = ipCheck.checkIPinRangeOrMatch(cidr,
+        ipCheck.parseIp(ip));
+    assert.strictEqual(actualRes, expectedRes);
+}
+
+function cidrListMatchCheck(cidrList, ip, expectedRes) {
+    const actualRes = ipCheck.ipMatchCidrList(cidrList, ip);
+    assert.strictEqual(actualRes, expectedRes);
+}
+
+describe('Parse IP address', () => {
+    it('should parse IPv4 address',
+        () => parseValidIpCheck('192.168.1.1', ipaddr.IPv4));
+
+    it('should parse IPv6 address',
+        () => parseValidIpCheck('2001:cdba::3257:9652', ipaddr.IPv6));
+
+    it('should parse IPv4 mapped IPv6 address',
+        // ::ffff:c0a8:101 mapped for 192.168.1.1
+        () => parseValidIpCheck('::ffff:c0a8:101', ipaddr.IPv4));
+
+    ['192.168', 'INVALID', '', null, undefined].forEach(item => {
```

Comment:
While not RFC'd (I think), 192.168 will parse correctly about everywhere to 192.0.0.168 (description here).

To have something invalid, I would suggest 260.384.2.1